bcda/web/api.go

```diff
@@ -305,12 +305,26 @@ func validateRequest(r *http.Request) ([]string, *fhirmodels.OperationOutcome) {
 	params, ok = r.URL.Query()["_since"]
 	if ok {
 		// Regex pattern for FHIR Instant format found at https://www.hl7.org/fhir/datatypes.html#primitive
-		instantRegex := `([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))`
-		validSince, err := regexp.MatchString(instantRegex, params[0])
-		if err != nil || !validSince {
+		instantRegex := `^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))$`
+		validSinceFormat, err := regexp.MatchString(instantRegex, params[0])
+		if err != nil || !validSinceFormat {
 			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.FormatErr, "Invalid date format supplied in _since parameter.  Date must be in FHIR Instant format.")
 			return nil, oo
 		}
+		sinceDate, err := time.Parse("2006-01-02T15:04:05.000-07:00", params[0])
```

Comment:
The FHIR Instant format supports a timestamp that looks like this, but also a timestamp that contains varying decimal places in the seconds field, or a `Z` in place of a timezone.  I think the parsing is failing if the timestamp contains a `Z`.

Reply:
Can use `time.RFC339` format here, maybe?
But still need to make sure the num decimal places is supported.

Reply:
You can also use `time.RFC3339Nano` It'll handle the millis for you.

Ex: https://play.golang.org/p/Gbe1aQ5eXs8

Reply:
Since the spec allows for unbounded number of subsecond digits, we'll be able to handle micros/nanos if the user provided that (if we use RFC3339Nano)